packages/plugin-ext/src/hosted/node/hosted-plugin-localization-service.ts

```diff
@@ -102,7 +99,7 @@ export class HostedPluginLocalizationService implements BackendApplicationContri
         if (plugin.contributes?.localizations) {
             for (const localization of plugin.contributes.localizations) {
                 for (const translation of localization.translations) {
-                    const l10n = getL10nTranslation(translation);
+                    const l10n = await getL10nTranslation(plugin.metadata.model.packageUri, translation);
```

Comment:
Could this be improved by collecting all `getL10nTranslations` as promises and running them together?

If there is no dependency between `plugin.contributes?.localizations` and `plugin.metadata.model.l10n` async tasks, maybe running them all together could be an additional improvement. Assuming they're IO (file read)